Grbl_Esp32/src/Stepper.cpp

```diff
@@ -414,26 +324,26 @@ static void stepper_pulse_func() {
         }
     }
 
-#ifdef USE_I2S_STEPS
-    if (current_stepper == ST_I2S_STREAM) {
-        //
-        // Generate pulse (at least one pulse)
-        // The pulse resolution is limited by I2S_OUT_USEC_PER_PULSE
-        //
-        i2s_out_push_sample(pulse_microseconds->get() / I2S_OUT_USEC_PER_PULSE);
-        set_stepper_pins_on(0);  // turn all off
-        return;
-    }
-#endif
-#ifdef USE_RMT_STEPS
-    return;
-#else
-    // wait for step pulse time to complete...some of it should have expired during code above
-    while (esp_timer_get_time() - step_pulse_start_time < pulse_microseconds->get()) {
-        NOP();  // spin here until time to turn off step
+    switch (current_stepper) {
+        case ST_I2S_STATIC:
+        case ST_I2S_STREAM:
+            //
+            // Generate pulse (at least one pulse)
+            // The pulse resolution is limited by I2S_OUT_USEC_PER_PULSE
+            //
+            i2s_out_push_sample(pulse_microseconds->get() / I2S_OUT_USEC_PER_PULSE);
```

Comment:
Currently, `I2S_OUT_USEC_PER_PULSE` is only valid if `USE_I2S_OUT` is defined, as is the prototype for the i2s_out function.
We need to conditionally enable the entire i2s_out_push_sample() in `#ifdef USE_I2S_OUT`,
or change the definition of I2SOut.h so that I2S_OUT_USEC_PER_PULSE and pro-type declarations are always valid.